### ISSUE TITLE
Workspace is crashing 

### ISSUE THREAD

**Issue by samyak-anand** [microsoft/vscode#216077]
Type: <b>Performance Issue</b>

I am facing issue, that the VCS is crashing at irregural time interval. It is also impacting the my work . Autosave is not working. Also,  the extentions also stop working. 
I am facing these issue from last 2 weeks. 

VS Code version: Code 1.90.1 (611f9bfce64f25108829dd295f54a6894e87339d, 2024-06-11T21:02:43.666Z)
OS version: Linux x64 6.5.0-41-generic snap
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i3-8130U CPU @ 2.20GHz (4 x 3399)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|5, 4, 2|
|Memory (System)|11.41GB (6.86GB free)|
|Process Argv|--force-user-env --no-sandbox --use-gl=angle --use-angle=swiftshader --crash-reporter-id b87051d2-06f3-4cf7-bed5-cf2c0d930d7a|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|wayland|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    0	   175	  8795	code main
    0	    35	  8797	   zygote
    0	    35	  8798	   zygote
    0	   210	  8871	window [1] (● Unit1.ipynb - Deep_Learning_crash_course - Visual Studio Code)
    0	   117	  8924	     window
    1	    94	  9254	window [2] (Issue Reporter)
    6	   164	  8826	   gpu-process
    0	    58	  8834	   utility-network-service
    0	   257	  8903	extensionHost [1]
    0	   374	  8979	     /home/samyak/.vscode/extensions/sourcery.sourcery-1.19.0/sourcery_binaries/install/linux/sourcery lsp
    0	    58	  9089	     electron-nodejs (e.js )
   15	   187	  9096	     electron-nodejs (bundle.js )
    0	    94	  8936	shared-process
    0	     0	  9265	     /bin/sh -c /usr/bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	   105	  8937	fileWatcher [1]
    0	    82	  9009	ptyHost
    0	     0	  9029	     /usr/bin/bash --init-file /snap/code/161/usr/share/code/resources/app/out/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh
    0	     0	  9081	     /usr/bin/bash --init-file /snap/code/161/usr/share/code/resources/app/out/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (● Unit1.ipynb - Deep_Learning_crash_course - Visual Studio Code)
|    Folder (Deep_Learning_crash_course): more than 20527 files
|      File types: py(6504) pyc(6504) pyi(2113) h(320) so(132) txt(98) afm(60)
|                  c(58) typed(55) f90(49)
|      Conf files:;
```

</details>
<details><summary>Extensions (23)</summary>

Extension|Author (truncated)|Version
---|---|---
project-manager|ale|12.8.0
gitignore|cod|0.9.0
python-snippets|cst|0.1.2
python-extensions-pack|dem|1.0.3
python-preview|don|0.0.4
git-extension-pack|don|0.1.3
githistory|don|0.6.20
gitlens|eam|15.1.0
python-resource-monitor|kai|0.2.3
vsc-python-indent|Kev|1.18.0
debugpy|ms-|2024.6.0
python|ms-|2024.8.1
vscode-pylance|ms-|2024.6.1
jupyter|ms-|2024.5.0
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.18
vscode-jupyter-cell-tags|ms-|0.1.9
vscode-jupyter-slideshow|ms-|0.1.6
vscode-python-typehint|njq|1.5.1
indent-rainbow|ode|8.3.1
sourcery|sou|1.19.0
livecode|xir|1.3.10
vscode-open-in-github|ziy|1.3.6

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
vscaat:30438848
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
jchc7451:31067544
chatpanelt:31048053
dsvsc021:30996838
f3je6385:31013174
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
pythonprt:31056678
dwnewjupytercf:31046870
newcmakeconfigv2:31071590

```

</details>

**Comment by gjsjohnmurray** [comment]
Maybe https://github.com/microsoft/vscode/issues/212494#issuecomment-2168893148 will help. Or wait for 1.90.2

**Comment by deepak1556** [comment]
Merging to https://github.com/microsoft/vscode/issues/212494, should be fixed in `1.90.2`
